substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/serialize/SerializationSupport.java

```diff
@@ -0,0 +1,222 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.reflect.serialize;
+
+import com.oracle.graal.pointsto.BigBang;
+import com.oracle.svm.core.configure.SerializationKey;
+import com.oracle.svm.core.jdk.serialize.SerializationRegistry;
+import com.oracle.svm.core.option.HostedOptionValues;
+import com.oracle.svm.core.option.RuntimeOptionValues;
+import com.oracle.svm.core.option.SubstrateOptionsParser;
+import com.oracle.svm.core.util.VMError;
+import com.oracle.svm.hosted.FeatureImpl;
+import com.oracle.svm.hosted.NativeImageOptions;
+import com.oracle.svm.reflect.serialize.hosted.SerializationFeature.SerializationFeatureOptions;
+import jdk.internal.reflect.ConstructorAccessor;
+import jdk.internal.reflect.ReflectionFactory;
+import org.graalvm.nativeimage.Platform;
+import org.graalvm.nativeimage.Platforms;
+import org.graalvm.nativeimage.hosted.Feature.FeatureAccess;
+
+import java.io.Externalizable;
+import java.io.ObjectStreamClass;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class SerializationSupport implements SerializationRegistry {
+    // Cached SerializationConstructorAccessors for runtime usage
+    private Map<String, Map<String, Object>> cachedSerializationConstructorAccessors;
+    private static final String MULTIPLE_CHECKSUMS = "MULTIPLE_CHECKSUM";
+    private static final String CHECKSUM_VERIFY_FAIL = "CHECKSUM_VERIFY_FAIL";
+    /*
+     * This map is used to track multiple classloader usage. Map key is serialization target class
+     * name, value is the class name and class hierarchy checksum
+     */
+    private Map<String, List<SerializationKey>> serializationTargetAndChecksums;
```

Comment:
I don't really see the need to track these duplicates and checksums separately -- instead, the only error should be a mismatching checksum, and we can just record these as messages when we see them and submit to `bb.getUnsupportedFeatures()`.

Reply:
Using `serializationTargetAndChecksums` can keep all error messages at one place when there are more than two checksum mismatches (although is rare)

Reply:
fixed